app/controllers/api/v0/market_vendors_controller.rb

```diff
@@ -1,11 +1,15 @@
 class Api::V0::MarketVendorsController < ApplicationController
+
   def create
-    market_vendor = MarketVendor.new(market_vendor_params)
-    if market_vendor.save
-      render json: { message: "Successfully added vendor to market" }, status: :created
-    else
-      message = market_vendor.errors.messages
-      render json: ErrorSerializer.new(ErrorMessage.new(message, 404)).serializer_validation, status: :not_found
+    check_nil_values
+    # This method helps us raise the RecordInvalid, 400 response we want for missing ids
+    market = Market.find(params[:market_id])
+    vendor = Vendor.find(params[:vendor_id])
+    # These two methods help us raise the RecordNotFound, 404 response we want for ids passed that don't exist
```

Comment:
Thank you for bringing up that find will throw an error I spent forever trying to figure that out and then I remember what you said earlier